jvm-agent/src/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckClassVisitor.kt

```diff
@@ -291,81 +232,91 @@ internal class LincheckClassVisitor(
+        if (methodName == "<clinit>")
+            return true
+        // Wrap `ClassLoader::loadClass(className)` calls into ignored sections
+        // to ensure their code is not analyzed by the Lincheck.
+        if (isClassLoaderClassName(className.toCanonicalClassName()) && isLoadClassMethod(methodName, descriptor))
+            return true
+        // Wrap `MethodHandles.Lookup.findX` and related methods into ignored sections
+        // to ensure their code is not analyzed by the Lincheck.
+        if (isIgnoredMethodHandleMethod(className.toCanonicalClassName(), methodName))
+            return true
+        // Wrap all methods of the ` StackTraceElement ` class into ignored sections.
+        // Although `StackTraceElement` own bytecode should not be instrumented,
+        // it may call functions from `java.util` classes (e.g., `HashMap`),
+        // which can be instrumented and analyzed.
+        // At the same time, `StackTraceElement` methods can be called almost at any point
+        // (e.g., when an exception is thrown and its stack trace is being collected),
+        // and we should ensure that these calls are not analyzed by Lincheck.
+        //
+        // See the following issues:
+        //   - https://github.com/JetBrains/lincheck/issues/376
+        //   - https://github.com/JetBrains/lincheck/issues/419
+        if (isStackTraceElementClass(className.toCanonicalClassName()))
+            return true
+        // Ignore methods of JDK 20+ `ThreadContainer` classes, except `start` method.
+        if (isThreadContainerClass(className.toCanonicalClassName()) &&
+            !isThreadContainerThreadStartMethod(className.toCanonicalClassName(), methodName))
+            return true
+
+        return false
     }
 
-    protected fun isKnownLineNumber(): Boolean = lineNumber > 0
-
-    override fun visitLineNumber(line: Int, start: Label) {
-        lineNumber = line
-        super.visitLineNumber(line, start)
-    }
-}
-
-private class WrapMethodInIgnoredSectionTransformer(
-    fileName: String,
-    className: String,
-    methodName: String,
-    adapter: GeneratorAdapter,
-) : ManagedStrategyMethodVisitor(fileName, className, methodName, adapter) {
-
-    private val tryBlock: Label = adapter.newLabel()
-    private val catchBlock: Label = adapter.newLabel()
-
-    override fun visitCode() = adapter.run {
-        visitCode()
-        invokeStatic(Injections::enterIgnoredSection)
-        visitTryCatchBlock(tryBlock, catchBlock, catchBlock, null)
-        visitLabel(tryBlock)
+    private fun shouldNotInstrument(className: String, methodName: String, descriptor: String): Boolean {
+        // Do not instrument `ClassLoader` methods.
+        if (isClassLoaderClassName(className.toCanonicalClassName()))
+            return true
+        // Instrumentation of `java.util.Arrays` class causes some subtle flaky bugs.
+        // See details in https://github.com/JetBrains/lincheck/issues/717.
```

Comment:
Must be the intrinsics problem, some classes which do not have `@IntrinsicCandidate` are still in fact intrinsified and delete out instrumentation mid-execution